source/MaterialXCore/Interface.cpp

```diff
@@ -366,8 +394,16 @@ NodePtr Input::getConnectedNode() const
     }
     if (hasNodeName())
     {
+        // Check on current parent, and graph parent if any
+        const string& nodeName = getNodeName();
         ConstGraphElementPtr graph = getAncestorOfType<GraphElement>();
-        NodePtr node = graph ? graph->getNode(getNodeName()) : nullptr;
+        NodePtr node = graph ? graph->getNode(nodeName) : nullptr;
```

Comment:
For a given nodegraph input check at the parent nodegraph level for a connection.